app/services/payment_provider_customers/stripe_service.rb

```diff
@@ -203,6 +203,21 @@ def deliver_error_webhook(stripe_error)
       )
     end
 
+    def fail!(stripe_error)
+      result.fail_with_error!(PaymentProviderFailure.new(
```

Comment:
To keep the "easy" way, you could rely on `result.provider_error!(...)`. That way, no need to build yourself the instance of the error at this stage